src/test/java/bio/terra/pact/provider/SnapshotsApiControllerTest.java

```diff
@@ -0,0 +1,118 @@
+import au.com.dius.pact.provider.junitsupport.Provider;
+import au.com.dius.pact.provider.junitsupport.State;
+import au.com.dius.pact.provider.junitsupport.loader.PactBroker;
+import au.com.dius.pact.provider.junitsupport.loader.PactBrokerConsumerVersionSelectors;
+import au.com.dius.pact.provider.junitsupport.loader.SelectorBuilder;
+import au.com.dius.pact.provider.spring.junit5.MockMvcTestTarget;
+import au.com.dius.pact.provider.spring.junit5.PactVerificationSpringProvider;
+import bio.terra.app.controller.GlobalExceptionHandler;
+import bio.terra.app.controller.SnapshotsApiController;
+import bio.terra.common.category.Pact;
+import bio.terra.common.exception.ForbiddenException;
+import bio.terra.common.iam.AuthenticatedUserRequestFactory;
+import bio.terra.model.SnapshotModel;
+import bio.terra.model.SnapshotRetrieveIncludeModel;
+import bio.terra.model.TableModel;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.dataset.AssetModelValidator;
+import bio.terra.service.dataset.IngestRequestValidator;
+import bio.terra.service.filedata.FileService;
+import bio.terra.service.job.JobService;
+import bio.terra.service.snapshot.SnapshotRequestValidator;
+import bio.terra.service.snapshot.SnapshotService;
+import bio.terra.service.snapshot.exception.SnapshotNotFoundException;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+
+/**
+ * Establishing SnapshotApiController Provider states via mocking for use in verifying contracts
+ * made against these APIs by external Consumers.
+ */
+@Tag(Pact.TAG)
+@ActiveProfiles(Pact.PROFILE)
+@Provider(Pact.PACTICIPANT)
+@PactBroker
+@ContextConfiguration(classes = {SnapshotsApiController.class, GlobalExceptionHandler.class})
+@WebMvcTest
+class SnapshotsApiControllerTest {
+
+  @Autowired private MockMvc mvc;
+
+  @MockBean private JobService jobService;
+  @MockBean private SnapshotRequestValidator snapshotRequestValidator;
+  @MockBean private SnapshotService snapshotService;
+  @MockBean private IamService iamService;
+  @MockBean private IngestRequestValidator ingestRequestValidator;
+  @MockBean private FileService fileService;
+  @MockBean private AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  @MockBean private AssetModelValidator assetModelValidator;
+
+  @PactBrokerConsumerVersionSelectors
+  public static SelectorBuilder consumerVersionSelectors() {
```

Comment:
Nice...I'm totally going to copy this.

Do you know what happens when you _don't_ specify this?  Is what's defined here also the default?

Reply:
All credit to @ichengchang who put up a draft PR to model this selection!  I asked that same question in a comment:

https://github.com/DataBiosphere/jade-data-repo/pull/1534#discussion_r1385422359

My most pressing issue at the time was that since TDR had no Pacts published to it from a consumer's **default** branch, TDR Pact verification was reporting no Pacts found.  I could see Pacts published by WDS against TDR, but they weren't being picked up because they were being published by a WDS feature branch.